lib/reverse_sentence.rb

```diff
@@ -1,6 +1,28 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  if my_sentence.nil?
+    return nil
+  end
+  
+  word_array = my_sentence.split(/\.?\s+/)
```

Comment:
You are splitting the sentence into a new array, which you can avoid to reduce space complexity.